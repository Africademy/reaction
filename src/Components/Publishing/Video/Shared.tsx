import { Row } from 'react-styled-flexboxgrid'
import styled from "styled-components"

export const MaxRow = styled(Row) `
  max-width: 1200px;
  margin: 0px auto;
  padding: 0px 12px;
`
