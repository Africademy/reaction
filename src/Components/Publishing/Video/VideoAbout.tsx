import React, { Component } from "react"
import { Col, Row } from 'react-styled-flexboxgrid'
import styled from "styled-components"
import { pMedia } from "../../Helpers"
import { Fonts } from "../Fonts"
import { Text } from '../Sections/Text'

interface Props {
  article: any
}

export class VideoAbout extends Component<Props, null> {
  render() {
    const { article } = this.props
    const { media } = article
    return (
      <VideoAboutRow>
        <Col
          xs={12}
          sm={4}
          first="xs"
          last="sm"
        >
          <Title>
            Credits
          </Title>
          <Text layout="standard" html={media.credits} />
        </Col>

        <Col xs={12} sm={8} md={8} lg={8}>
          <Title>
            About the Film
          </Title>
          <Text layout="standard" html={media.description} />
        </Col>
      </VideoAboutRow>
    )
  }
}

const Title = styled.div`
  margin-bottom: 15px;
  ${Fonts.unica("s34")}
  ${pMedia.md`
    ${Fonts.unica("s32")}
  `}
`
const VideoAboutRow = styled(Row)`
  max-width: 1200px;
  margin: 60px auto;
`
