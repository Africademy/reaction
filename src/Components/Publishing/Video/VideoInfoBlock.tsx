import { Sans } from "@artsy/palette"
import { unica } from "Assets/Fonts"
import React, { Component } from "react"
import { Row } from "react-styled-flexboxgrid"
import styled from "styled-components"
import { formatTime } from "../Constants"

interface Props {
  editTitle?: any
  media?: any
  subTitleLink?: string
  subTitle?: string
  title?: string
}

export class VideoInfoBlock extends Component<Props, null> {
  render() {
    const { editTitle, media, subTitle, subTitleLink, title } = this.props

    return (
      <div>
        <Row>
          {subTitle && (
            <SubTitle size="3t">
              {subTitleLink ? <a href={subTitleLink}>{subTitle}</a> : subTitle}
            </SubTitle>
          )}
          <Sans size="3t">{formatTime(media.duration)}</Sans>
        </Row>
        <Row>
          <MediaTitle>{editTitle || title}</MediaTitle>
        </Row>
      </div>
    )
  }
}

const SubTitle = Sans.extend`
  margin-right: 35px;

  a {
    color: white;
    text-decoration: none;
  }
`

const MediaTitle = styled.span`
  position: relative;
  ${unica("s45")};
`
