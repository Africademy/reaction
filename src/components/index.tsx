import Artwork from "./artwork"
import ArtworkFilter from "./artwork_filter"
import Buttons from "./buttons/index"
import Modal from "./modal/modal"

import ArtworkGrid from "./artwork_grid"
import BorderedPulldown from "./bordered_pulldown"
import Checkbox from "./checkbox"
import Grid from "./grid"
import Icon from "./icon"
import Input from "./input"
import Nav from "./nav"
import Text from "./text"
import TextArea from "./text_area"
import TextLink from "./text_link"
import Title from "./title"

export default {
  Artwork,
  ArtworkFilter,
  Buttons,
  Modal,
  ArtworkGrid,
  BorderedPulldown,
  Checkbox,
  Grid,
  Icon,
  Input,
  Nav,
  Text,
  TextArea,
  TextLink,
  Title,
}
