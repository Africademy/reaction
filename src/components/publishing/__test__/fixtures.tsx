import { Props as ImageSetPreviewProps } from "../imageset_preview"

export const Artworks = [
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
    },
    artist: {
      name: "Fernando Botero",
    },
    width: 1152,
    height: 826,
  },
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
      slug: "gary-nader",
    },
    artists: [
      {
        name: "Fernando Botero",
        slug: "fernando-botero",
      },
      {
        name: "Frida Kahlo",
        slug: "frida-kahlo",
      },
    ],
    width: 1152,
    height: 826,
  },
]

export const Articles = [
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "classic",
    vertical: null,
  },
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "standard",
    vertical: {
      name: "Art Market",
      id: "12345",
    },
  },
  {
    title: "New York's Next Art District",
    contributing_authors: [
      {
        name: "Molly Gottschalk",
        id: "12345",
      },
    ],
    published_at: "2017-05-19T13:09:18.567Z",
    layout: "feature",
    vertical: {
      name: "Art Market",
      id: "12345",
    },
  },
]

export const Authors = [
  {
    image_url: "https://artsy-media-uploads.s3.amazonaws.com/9vcX6FR21rKHatmvJ8K0sg%2FAbigail.jpg",
    name: "Abigail Cain",
    bio: "[Abigail Cain](https://artsy.net/abigail) is an Art Market Editor at Artsy",
    twitter_handle: "abigailcain",
  },
  {
    image_url: "http://files.artsy.net/images/molly_01.jpg",
    name: "Molly Gottschalk",
    bio:
      "[Molly Gottschalk](https://artsy.net/molly) is a Features Producer at Artsy. She is a photographer and lover of cats.",
    twitter_handle: "mollygottschalk",
  },
  {
    name: "Halley Johnson",
    bio: "[Halley Johnson](https://artsy.net/molly) is an Writer Guru at Artsy. She enjoys speaking Georgian.",
    twitter_handle: "halleyjohnson",
  },
]

export const Embeds = [
  {
    mobile_height: 1300,
    height: 1000,
    url: "http://files.artsy.net/documents/1parrasch.html",
    layout: "overflow",
    type: "embed",
  },
]

export const HeroSections = [
  {
    type: "text",
    url: "https://artsy-media-uploads.s3.amazonaws.com/YqTtwB7AWqKD95NGItwjJg%2FRachel_Rossin_portrait_2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "split",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "fullscreen",
    url: "https://artsy-media-uploads.s3.amazonaws.com/ZR0wtJhg5Nez7Vm8uCP8Nw%2FDSC_0720-Edit-2.jpg",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "split",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "fullscreen",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
  {
    type: "text",
    url: "https://artsy-media-uploads.s3.amazonaws.com/z9w_n6UxxoZ_u1lzt4vwrw%2FHero+Loop+02.mp4",
    subheader: "Nulla vitae elit libero, a pharetra augue. Vestibulum id ligula porta felis euismod semper.",
  },
]

export const Images = [
  {
    type: "artwork",
    id: "589a6291275b2410d1beb6a5",
    slug: "fernando-botero-nude-on-the-beach",
    date: "2000",
    title: "Nude on the Beach",
    image: "https://d32dm0rphc51dk.cloudfront.net/0aRUvnVgQKbQk5dj8xcCAg/larger.jpg",
    partner: {
      name: "Gary Nader",
      slug: "gary-nader",
    },
    artists: [
      {
        name: "Fernando Botero",
        slug: "fernando-botero",
      },
    ],
    artist: {
      name: "Fernando Botero",
      slug: "fernando-botero",
    },
    width: 1152,
    height: 826,
  },
  {
    url: "https://artsy-media-uploads.s3.amazonaws.com/co8j2xq40ROMyBrJQm_4eQ%2FDafenOilPaintingVillage_AK03.jpg",
    type: "image",
    width: 900,
    height: 1200,
    caption: "<p>Photo by Adam Kuehl for Artsy.</p>",
  },
  {
    url: "https://d32dm0rphc51dk.cloudfront.net/CpHY-DRr7KW0HGXLslCXHw/larger.jpg",
    type: "image",
    width: 816,
    height: 1024,
    caption: "<p>Photo by Adam Kuehl for Artsy. Image courtesy of the Guggenheim Museum.</p>",
  },
]

export const ImageSetFull: ImageSetPreviewProps["section"] = {
  type: "image_set",
  layout: "full",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const ImageSetMini: ImageSetPreviewProps["section"] = {
  type: "image_set",
  title: "The Work of Bruce M. Sherman",
  images: Images,
}

export const ImageSetFullSansTitle: ImageSetPreviewProps["section"] = {
  type: "image_set",
  layout: "full",
  images: Images,
}

export const ImageSetMiniSansTitle: ImageSetPreviewProps["section"] = {
  type: "image_set",
  images: Images,
}
export const Videos = [
  {
    url: "https://www.youtube.com/watch?v=PXi7Kjlsz9A",
    caption: "<p>What motivates patrons to fund artists’ wildest dreams?</p>",
    cover_image_url: "https://artsy-media-uploads.s3.amazonaws.com/IB6epb5L_l0rm9btaDsY7Q%2F14183_MDP_Evening_240.jpg",
  },
  {
    url: "https://vimeo.com/191988155",
    caption:
      "<p>2016 was a memorable year for the world, and art along with it. Powered by data culled from Artsy as well as UBS’s Planet Art app, “The Year in Art 2016” will explore how the creative community responded to the cultural shifts and tribulations this year has seen—from the destruction of Palmyra to the proliferation of Virtual Reality to the U.S. election.</p>",
    cover_image_url: "https://artsy-media-uploads.s3.amazonaws.com/ditbyaUgdcl6mHin07TfKA%2FMassimilianoGioni_0581.jpg",
  },
  {
    url: "https://vimeo.com/191988155",
  },
]
