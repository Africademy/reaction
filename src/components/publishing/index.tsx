import Article from "./article"
import Header from "./header/header"
import Artwork from "./sections/artwork"
import Authors from "./sections/authors"
import Embed from "./sections/embed"
import Image from "./sections/image"
import ImageCollection from "./sections/image_collection"
import ImagesetPreview from "./sections/imageset_preview"
import ImagesetPreviewClassic from "./sections/imageset_preview_classic"
import Video from "./sections/video"

export default {
  Article,
  Artwork,
  Authors,
  Embed,
  Header,
  Image,
  ImageCollection,
  ImagesetPreview,
  ImagesetPreviewClassic,
  Video,
}
