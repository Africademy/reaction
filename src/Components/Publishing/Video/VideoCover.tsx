import React, { Component } from "react"
import { Col } from 'react-styled-flexboxgrid'
import styled, { StyledFunction } from "styled-components"
import { media } from "../../Helpers"
import { Fonts } from "../Fonts"
import { IconVideoPlay } from "../Icon/IconVideoPlay"
import { MaxRow } from "./Shared"
import { VideoInfoBlock } from "./VideoInfoBlock"

interface Props {
  media?: any
  seriesTitle?: string
  description?: string
  playVideo?: () => void
  hideCover?: boolean
}

interface CoverProps {
  hideCover?: boolean
}

interface CoverAssetProps {
  src?: string
}

export class VideoCover extends Component<Props, null> {
  render() {
    const {
      description,
      media,
      seriesTitle,
      playVideo,
      hideCover
    } = this.props

    return (
      <VideoCoverContainer hideCover={hideCover}>
        <VideoCoverAsset src={media.cover_image_url} />
        <VideoCoverOverlay />
        <VideoCoverInfo>
          <VideoCoverInfoRow>
            <Col xs={2} sm={1} onClick={playVideo}>
              <IconVideoPlay color="white" />
            </Col>
            <Col xs={10} sm={3}>
              <VideoInfoBlock
                media={media}
                seriesTitle={seriesTitle}
              />
            </Col>
            <Col xs={12} sm={8}>
              <MediaDescription>
                {description}
              </MediaDescription>
            </Col>
          </VideoCoverInfoRow>
        </VideoCoverInfo>
      </VideoCoverContainer>
    )
  }
}

const Div: StyledFunction<CoverAssetProps> = styled.div
export const VideoCoverAsset = Div`
  background: url(${props => props.src || ""}) no-repeat center center;
  background-size: cover;
  background-color: black;
`

const VideoCoverOverlay = styled.div`
  opacity: 0.75;
  background-color: black;
`

const VideoCoverInfoRow = styled(MaxRow)`
  align-items: flex-end;
`

const VideoCoverInfo = styled.div`
  display: flex;
  align-items: flex-end;
  box-sizing: border-box;
  padding-bottom: 60px;
  z-index: 1;
  ${IconVideoPlay} {
    height: 60px;
    width: 44px;
    margin-right: 15px;
  }
  ${media.sm`
    padding-bottom: 40px;
  `}
`

const CoverDiv: StyledFunction<CoverProps> = styled.div
export const VideoCoverContainer = CoverDiv`
  position: relative;
  width: 100vw;
  height: 100vh;
  color: white;
  opacity: ${props => props.hideCover ? '0' : '1'};
  visibility: ${props => props.hideCover ? 'hidden' : 'visible'};
  transition: opacity 0.5s ease visibility 0.5s ease;
  ${VideoCoverAsset}, ${VideoCoverOverlay}, ${VideoCoverInfo} {
    position: absolute;
    width: 100%;
    height: 100%;
  }
`

const MediaDescription = styled.div`
  ${Fonts.garamond("s23")}
  ${media.sm`
    ${Fonts.garamond("s19")}
    margin-top: 30px;
  `}
`
