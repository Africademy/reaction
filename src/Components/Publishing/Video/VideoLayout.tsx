import React, { Component } from "react"
import { Col } from 'react-styled-flexboxgrid'
import styled from "styled-components"
import { media } from "../../Helpers"
import { getEditorialHref } from "../Constants"
import { Fonts } from "../Fonts"
import { Nav } from "../Nav/Nav"
import { ArticleCard, ArticleCardContainer } from "../Series/ArticleCard"
import { SeriesAbout, SeriesAboutContainer } from "../Series/SeriesAbout"
import { VideoContainer, VideoPlayer } from "../VideoPlayer/VideoPlayer"
import { MaxRow } from "./Shared"
import { VideoAbout, VideoAboutContainer } from "./VideoAbout"
import { VideoCover } from "./VideoCover"

interface Props {
  article: any
  seriesArticle?: any
  relatedArticles?: any
}

interface State {
  isPlaying: boolean
}

export class VideoLayout extends Component<Props, State> {
  state = {
    isPlaying: false
  }

  playVideo = () => {
    this.setState({
      isPlaying: true
    })
  }

  onPauseVideo = () => {
    this.setState({
      isPlaying: false
    })
  }

  render() {
    const {
      article,
      seriesArticle,
      relatedArticles
    } = this.props
    const { media } = article

    return(
      <VideoLayoutContainer>
        <Nav
          transparent
          sponsor={article.sponsor}
        />
        <VideoPlayerContainer>
          <VideoPlayer
            url={media.url}
            title={media.title}
            forcePlay={this.state.isPlaying}
            notifyIsPaused={this.onPauseVideo}
          />
          <VideoCover
            media={media}
            seriesTitle={seriesArticle && seriesArticle.title}
            description={article.description}
            playVideo={this.playVideo}
            hideCover={this.state.isPlaying}
          />
        </VideoPlayerContainer>
        <MaxRow>
          <VideoAbout
            article={article}
            color="white"
          />
        </MaxRow>
        {relatedArticles &&
          <MaxRow>
            <RelatedArticlesTitle>
              {"More in "}
              {seriesArticle ?
                <Link href={getEditorialHref('series', seriesArticle.slug)}>
                  {seriesArticle.title}
                </Link>
                :
                <span>{article.vertical && article.verical.name}</span>
              }
            </RelatedArticlesTitle>
          </MaxRow>
        }
        {relatedArticles &&
          relatedArticles.map((relatedArticle, i) => {
            return (
              <MaxRow key={i}>
                <Col xs={12}>
                  <ArticleCard
                    article={relatedArticle}
                    color="white"
                    series={seriesArticle}
                  />
                </Col>
              </MaxRow>
            )
          })
        }
        {seriesArticle &&
          <MaxRow>
            <SeriesAbout
              article={seriesArticle}
              color="white"
            />
          </MaxRow>
        }
      </VideoLayoutContainer>
    )
  }
}

const RelatedArticlesTitle = styled(Col) `
  ${Fonts.unica("s32")}
`

const VideoLayoutContainer = styled.div`
  background: black;
  color: white;
  margin: auto;

  ${Nav} {
    position: absolute;
    top: 0;
  }

  ${ArticleCardContainer} {
    margin-bottom: 60px;
  }

  ${RelatedArticlesTitle} {
    margin-bottom: 40px;
  }

  ${SeriesAboutContainer}, ${VideoAboutContainer} {
    margin: 60px 0 100px 0;
  }

  ${media.sm`
    ${ArticleCardContainer} {
      margin-bottom: 30px;
    }
    ${SeriesAboutContainer}, ${VideoAboutContainer} {
      margin: 40px 0 100px 0;
    }
  `}
`

const VideoPlayerContainer = styled.div`
  position: relative;
  width: 100vw;
  height: 100vh;

  ${VideoContainer} {
    position: absolute;
    top: 0;
  }
`

const Link = styled.a`
  text-decoration: none;
  color: white;
`
