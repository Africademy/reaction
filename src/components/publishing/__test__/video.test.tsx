import * as React from "react"
import * as renderer from "react-test-renderer"

import "jest-styled-components"

import Video from "../sections/video"
import { Videos } from "./fixtures/components"

jest.mock("react-sizeme", () => jest.fn(c => d => d))

it("renders properly", () => {
  const video = renderer.create(<Video section={Videos[0]} />).toJSON()
  expect(video).toMatchSnapshot()
})
