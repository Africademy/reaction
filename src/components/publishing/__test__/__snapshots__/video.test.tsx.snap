// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="file__VideoContainer-s5ax5tw-0 fTLvxj"
>
  <div
    className="sc-htpNat UgtSs"
    height={281.25}
    hidden={false}
    onClick={[Function]}
    src="https://artsy-media-uploads.s3.amazonaws.com/IB6epb5L_l0rm9btaDsY7Q%2F14183_MDP_Evening_240.jpg"
  >
    <div
      className="file__PlayButton-s5ax5tw-2 dFWWEK"
    >
      <div
        className="file__PlayButtonCaret-s5ax5tw-1 bWSLbp"
      />
    </div>
  </div>
  <iframe
    allowFullScreen={true}
    className="sc-bwzfXH bSAkii"
    frameBorder="0"
    height={281.25}
    src="https://www.youtube.com/embed/PXi7Kjlsz9A?title=0&portrait=0&badge=0&byline=0&showinfo=0&rel=0&controls=2&modestbranding=1&iv_load_policy=3&color=E5E5E5"
  />
  <div
    className="file__CaptionContainer-q52b23-0 ikvQHr"
  >
    <div
      className="sc-bdVaJa bGfehm"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>What motivates patrons to fund artistsâ€™ wildest dreams?</p>",
        }
      }
    />
  </div>
</div>
`;
